src/engine/math_util.c

```diff
@@ -273,25 +273,45 @@ void mtxf_lookat(Mat4 mtx, Vec3f from, Vec3f to, s16 roll) {
 }
 
 /**
- * Set 'dest' to a transformation matrix that turns an object to face the camera.
+ * Set 'dest' to a transformation matrix that turns an object to face the camera using either spherical or cylindrical billboarding.
  * 'mtx' is the look-at matrix from the camera.
+ * 'camera' is the camera's look direction, only used for cylindrical billboarding
+ * 'axis' is the fixed axis around which the object will be rotated, if using cylindrical billboarding
  * 'position' is the position of the object in the world.
  * 'scale' is the scale of the object.
  * 'angle' rotates the object while still facing the camera.
+ * 'isCylindrical' determines which type of billboarding to use.
  */
-void mtxf_billboard(Mat4 dest, Mat4 mtx, Vec3f position, Vec3f scale, s16 angle) {
+void mtxf_billboard_generic(Mat4 dest, Mat4 mtx, Vec3f camera, Vec3f axis, Vec3f position, Vec3f scale, s16 angle, s32 isCylindrical) {
     PUPPYPRINT_ADD_COUNTER(gPuppyCallCounter.matrix);
-    s32 i;
-    f32 sx = scale[0];
-    f32 sy = scale[1];
-    f32 sz = scale[2];
-    Mat4* cameraMat = &gCameraTransform;
-    for (i = 0; i < 3; i++) {
-        for (int j = 0; j < 3; j++) {
-            dest[i][j] = (*cameraMat)[j][i];
+    register s32 i;
+    register f32 sx = scale[0];
+    register f32 sy = scale[1];
+    register f32 sz = scale[2];
```

Comment:
Nit: We removed all uses of the `register` keyword from math util because it doesn't actually do anything with our compiler flags.